commit bd41439fa696f58038b8e172379bddc742c4788f (HEAD, refs/heads/master)
Author:     P Purkayastha <ppurka@gmail.com>
AuthorDate: Wed Feb 5 13:08:58 2014 +0800
Commit:     P Purkayastha <ppurka@gmail.com>
CommitDate: Wed Feb 5 13:08:58 2014 +0800

    Patch 02/04: Remove --clean option
    
    This option is now completely redundant since we do not build
    in the source directory

diff --git a/easy_efl.sh b/easy_efl.sh
index 919560b..4b148e1 100644
--- a/easy_efl.sh
+++ b/easy_efl.sh
@@ -109,7 +109,6 @@ EASY_PWD=`pwd`
 accache=""
 easy_options=""
 command_options=$@
-clean=0
 mkdir -p "$HOME/.config/easy_efl" || {
     echo >&2 "ERROR: Can not create config directory $HOME/.config/easy_efl"
     exit 1
@@ -221,10 +220,6 @@ function logo ()
 					echo "      --srcpath=<path>                  = change the default source path"
 					echo "      --asuser                          = do everything as the user, not as root"
 					echo "      --no-sudopwd                      = sudo don't need a password..."
-					echo "  -c, --clean                           = clean the sources before building"
-					echo "                                          (more --cleans means more cleaning, up"
-					echo "                                          to a maximum of three, which will"
-					echo "                                          uninstall efl+enlightenment)"
 					echo "  -s, --skip-srcupdate                  = don't update sources"
 					echo "  -a, --ask-on-src-conflicts            = ask what to do with a conflicting"
 					echo "                                          source file"
@@ -460,39 +455,6 @@ function compile ()
 	rm -f "$status_path/$name.noerrors"
 	rm -f "$logs_path/$name.log"
 
-	if [ $clean -ge 1 ]; then
-		if [ -e "Makefile" ]; then
-			if [ $clean -eq 1 ]; then
-				run_command "$name" "$path" "clean" "clean  : " "$mode" "$make -j $threads clean"
-				if [ ! -e "$status_path/$name.noerrors" ]; then
-					if [ "$skip_errors" ]; then
-						write_name "$name" "hidden"	# clean might fail, that's ok
-					else
-						return
-					fi
-				fi
-			fi
-			if [ $clean -eq 2 ]; then
-				run_command "$name" "$path" "distclean" "distcln: " "$mode" "$make -j $threads clean distclean"
-				if [ ! -e "$status_path/$name.noerrors" ]; then
-					if [ "$skip_errors" ]; then
-						write_name "$name" "hidden"	# distclean might fail, that's ok
-					else
-						return
-					fi
-				fi
-			fi
-			if [ $clean -ge 3 ]; then
-				run_command "$name" "$path" "uninstall" "uninst : " "rootonly" "$make -j $threads uninstall clean distclean"
-				if [ ! -e "$status_path/$name.noerrors" ] ; then return ; fi
-
-			    # It's no longer installed if we just uninstalled it.
-			    # Even if the uninstall failed, it's best to mark it as uninstalled so that a partial uninstall gets fixed later.
-			    rm -f $status_path/$name.installed
-			fi
-		fi
-	fi
-
 	# get autogen arguments
 	args=""
 	for app_arg in `echo $autogen_args | tr -s '\,' ' '`; do
@@ -917,7 +879,6 @@ do
 		--srcpath)					src_path="$value" ;;
 		--asuser)					asuser=1 ;;
 		--no-sudopwd)				no_sudopwd=1 ;;
-		-c|--clean)					clean=$(($clean + 1))	;;
 		-d|--docs)					gen_docs=1 ;;
 		--postscript)				easy_efl_post_script="$value" ;;
 		-s|--skip-srcupdate)		skip_srcupdate=1 ;;
